tests/unit/test_charm.py

```diff
@@ -3,18 +3,55 @@
 #
 # Learn more about testing at: https://juju.is/docs/sdk/testing
 
-import unittest
-
 import ops
 import ops.testing
-from charm import OpenstackExporterOperatorCharm
+import pytest
+from charm import CLOUD_NAME, OS_CLIENT_CONFIG, SNAP_NAME, OpenstackExporterOperatorCharm
 
 
-class TestCharm(unittest.TestCase):
-    def setUp(self):
+class TestCharm:
+    def setup_method(self, _):
         self.harness = ops.testing.Harness(OpenstackExporterOperatorCharm)
-        self.addCleanup(self.harness.cleanup)
+
+    def teardown_method(self, _):
+        self.harness.cleanup()
+
+    @pytest.mark.parametrize(
+        "config",
+        [
+            ({"cache": False, "cache_ttl": "100s"}),
+            ({"cache": True, "cache_ttl": "200s"}),
+        ],
+    )
+    def test_config_changed(self, config, mocker):
+        mock_get_installed_snap_service = mocker.patch("charm.get_installed_snap_service")
+        mock_snap_service = mocker.Mock(spec_set=["configure", "restart_and_enable", "stop"])
```

Comment:
This should be `spec_set=SnapService`, there is no reason defining methods manually.

Reply:
Updated